src/utils/diagnostics.ts

```diff
@@ -11,14 +11,12 @@ export function isLogLevelEnabled(outputChannel: vscode.LogOutputChannel, logLev
     return outputChannel && outputChannel.logLevel > 0 && outputChannel.logLevel <= logLevel;
 }
 
-export function logEnvironment(outputChannel: vscode.LogOutputChannel, environment: NodeJS.ProcessEnv, title?: string): void {
+export function logProcessEnvironment(outputChannel: vscode.LogOutputChannel): void {
     if (isLogLevelEnabled(outputChannel, vscode.LogLevel.Debug)) {
         try {
-            if (title) {
-                outputChannel.debug(title);
-            }
+            outputChannel.debug(`--- Process Environment (${Object.getOwnPropertyNames(process.env).length}) ---`);
 
-            for (const key in environment) {
+            for (const key in process.env) {
```

Comment:
I think we'll want to do this, instead of `process.env`, since this is what gets used in VSCode terminals automatically, and in processes we launch thanks to `DefaultEnvStreamCommandRunnerFactory`:
`withDockerEnvSettings(process.env)`

Reply:
Also I think it needs to be `Object.keys(theThing)`

Reply:
I've broken the logging up by process environment and Docker environment from settings in order to clearly see where environment settings are coming from (see logProcessEnvironment and logDockerEnvironment) with the Docker environment being re-emitted if the change event fires.

I'll update to `for of Object.keys` to be safe and only echo own keys for the environment objects.